WordPress/src/main/java/org/wordpress/android/ui/jetpack/restore/RestoreViewModel.kt

```diff
@@ -0,0 +1,172 @@
+            // Show the next step only if it's a fresh activity so we can handle the navigation
+            wizardManager.showNextStep()
+        } else {
+            restoreState = requireNotNull(savedInstanceState.getParcelable(KEY_RESTORE_STATE))
+            val currentStepIndex = savedInstanceState.getInt(KEY_RESTORE_CURRENT_STEP)
+            wizardManager.setCurrentStepIndex(currentStepIndex)
+        }
+    }
+
+    fun addSnackbarMessageSource(snackbarEvents: LiveData<Event<SnackbarMessageHolder>>) {
+        _snackbarEvents.addSource(snackbarEvents) { event ->
+            _snackbarEvents.value = event
+        }
+    }
+
+    fun addErrorMessageSource(errorEvents: LiveData<Event<RestoreErrorTypes>>) {
+        _errorEvents.addSource(errorEvents) { event ->
+            _errorEvents.value = event
+        }
+    }
+
+    fun writeToBundle(outState: Bundle) {
+        outState.putInt(KEY_RESTORE_CURRENT_STEP, wizardManager.currentStep)
+        outState.putParcelable(KEY_RESTORE_STATE, restoreState)
+    }
+
+    fun onBackPressed() {
+        when (wizardManager.currentStep) {
+            DETAILS.id -> {
+                _wizardFinishedObservable.value = Event(RestoreCanceled)
+            }
+        }
+    }
+
+    private fun clearOldRestoreState(wizardStep: RestoreStep) {
+        if (wizardStep == DETAILS) {
+            restoreState = restoreState.copy(
+                    rewindId = null,
+                    restoreId = null,
+                    url = null,
+                    errorType = null
+            )
+        }
+    }
+
+    fun onRestoreDetailsFinished(rewindId: String?, restoreId: Long?, published: Date) {
+        restoreState = restoreState.copy(
+                rewindId = rewindId,
+                restoreId = restoreId,
+                published = published)
+        wizardManager.showNextStep()
+    }
+
+    fun onRestoreCanceled() {
+        _wizardFinishedObservable.value = Event(RestoreCanceled)
+    }
+
+    fun onRestoreProgressFinished(url: String?) {
+        restoreState = restoreState.copy(url = url)
+        wizardManager.showNextStep()
+    }
+
+    fun setToolbarState(toolbarState: ToolbarState) {
+        _toolbarStateObservable.value = toolbarState
+    }
+
+    fun transitionToError(errorType: RestoreErrorTypes) {
+        restoreState = restoreState.copy(errorType = errorType.id)
+        wizardManager.setCurrentStepIndex(RestoreStep.indexForErrorTransition())
+        wizardManager.showNextStep()
+    }
+
+    sealed class RestoreWizardState : Parcelable {
+        @Parcelize
+        object RestoreCanceled : RestoreWizardState()
+
+        @Parcelize
+        data class RestoreInProgress(val restoreId: Long) : RestoreWizardState()
+
+        @Parcelize
+        object RestoreCompleted : RestoreWizardState()
+    }
+
+    sealed class ToolbarState {
+        abstract val title: Int
+        abstract val icon: Int
+
+        data class DetailsToolbarState(
```

Comment:
1) Question (❓): This is currently unused. What if this is being used by the `RestoreDetailsViewModel`, just like it happens for the `BackupDownload` equivalent code. Will it be used as such in the future?
2) Question (❓): Since the `DetailsToolbarState` is added, I am guessing that more of that would become available later on in the future. If my assumption is correct, why not add all of them now even if the strings/icons are not known yet?

Reply:
1. It will be .. just the prep PR.
2. Trying to keep the pr's code relevant to specific topics. The other toolbar states will be added with each step implementation.